python/src/scipp/plot/plot_2d.py

```diff
@@ -182,72 +192,83 @@ def update_buttons(self, owner, event, dummy):
+        self.current_lims['y'] = extent_array[2:]
 
-        for xy, param in axparams.items():
+        # Slice data to slice down potential 3+D coords
+        self.slice_data()
+
+        for xy, param in self.axparams.items():
             # Create coordinate axes for resampled array to be used as image
             offset = 2 * (xy == "y")
             self.xyrebin[xy] = sc.Variable(
                 dims=[param["dim"]],
                 values=np.linspace(extent_array[0 + offset],
                                    extent_array[1 + offset],
                                    self.image_resolution[xy] + 1),
-                unit=self.slider_x[self.name][param["dim"]].unit)
+                unit=self.slider_coord[self.name][param["dim"]].unit)
+
+            # TODO: if labels are used on a 2D coordinates, we need to update
+            # the axes tick formatter to use xyrebin coords
 
             # Create bin-edge coordinates in the case of non bin-edges, since
             # rebin only accepts bin edges.
-            xydims = self.xyrebin[xy].dims
-            if not self.histograms[self.name][xydims[0]]:
-                self.xyedges[xy] = sc.Variable(
-                    dims=xydims,
-                    values=centers_to_edges(
-                        self.slider_x[self.name][xydims[0]].values),
-                    unit=self.slider_x[self.name][xydims[0]].unit)
+            if not self.histograms[self.name][param["dim"]][param["dim"]]:
+                # Account for the fact that some input data may not have any
+                # coordinates. Hence the slice has no coords and we need to
+                # get the coord from the slider_coord dict.
+                if param["dim"] not in self.vslice.coords:
+                    vcoord = self.slider_coord[self.name][param["dim"]]
+                else:
+                    vcoord = self.vslice.coords[param["dim"]]
+                dims = vcoord.dims
+                # Special handling for 2D coordinates
+                if len(self.slider_coord[self.name][param["dim"]].dims) > 1:
+                    shp = vcoord.shape
+                    idim = vcoord.dims.index(param["dim"])
+                    iother = (idim + 1) % 2
+                    shp[idim] += 1
+                    edges = sc.Variable(dims,
+                                        values=np.zeros(shp),
+                                        dtype=sc.dtype.float64)
+                    xory = dims[iother]
+                    for i in range(shp[iother]):
+                        edges[xory, i] = centers_to_edges(vcoord[xory,
+                                                                 i].values)
+                else:
+                    edges = sc.Variable(
+                        dims,
+                        values=centers_to_edges(
+                            self.slider_coord[self.name][param["dim"]].values),
+                        dtype=sc.dtype.float64)
```

Comment:
Why is special handling for 2D coords required here? What is different?

Reply:
So here I convert each row from center to edges.
The function `centers_to_edges` assumes it's getting a 1D array, so i can't send it the `.values` directly.

Reply:
Wouldn't it make more sense to fix `centers_to_edges`? Shouldn't be hard to write it for N-D data?

Reply:
yes, would probably make more sense to make `centers_to_edges` handle more dimensions.
Not sure it's super easy to write though...